src/execution/execute.js

```diff
@@ -353,6 +354,7 @@ function collectFields(
         if (!fields[name]) {
           fields[name] = [];
         }
+        selection.parentField = parentField;
```

Comment:
We want to ensure that we're not mutating the provided AST, as it may be used elsewhere.